components/ws-manager/pkg/manager/metrics.go

```diff
@@ -281,8 +306,6 @@ func (m *metrics) OnChange(status *api.WorkspaceStatus) {
 			return
 		}
 
-		removeFromState = true
-
```

Comment:
what's the reason for moving this to down below? If an early return gets triggered due to an error below, I think we end up leaking the status in `m.phaseState` as follows:

1. workspace goes into `STOPPED` phase
2. there's an error on line 327 or 338 and we return early. `removeFromState` is still `false`
3. the `defer` function updates `m.phaseState[status.Id]` to `STOPPED`
4. the next time `OnChange(status)` is triggered, we'll return early on line 305, as the previousState is STOPPED
5. `m.phaseState[status.Id]` doesn't get cleaned up